homeassistant/components/websocket_api/http.py

```diff
@@ -222,7 +222,7 @@ def handle_hass_stop(event):
                 self._to_write.put_nowait(None)
                 # Make sure all error messages are written before closing
                 await self._writer_task
-            except asyncio.QueueFull:
+            except (asyncio.QueueFull, asyncio.CancelledError):
```

Comment:
Instead of this, we should move up the changing of the data connections count to before we shut down the writer task.